tests/xml/blockly.xsd

```diff
@@ -0,0 +1,178 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
+    elementFormDefault="qualified"
+    targetNamespace="https://developers.google.com/blockly/xml"
+    xmlns="https://developers.google.com/blockly/xml">
+
+  <xs:element name="xml">
+    <xs:complexType>
+      <xs:choice>
+        <xs:sequence>
+          <xs:element ref="variables" minOccurs="0"/>
+          <xs:element ref="comment" minOccurs="0" maxOccurs="unbounded"/>
+          <xs:element ref="block" minOccurs="0" maxOccurs="unbounded"/>
+        </xs:sequence>
+        <xs:choice minOccurs="0" maxOccurs="unbounded">
+          <xs:element ref="category"/>
+          <xs:element ref="sep"/>
+        </xs:choice>
+      </xs:choice>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="category">
+    <xs:complexType>
+      <xs:choice minOccurs="0" maxOccurs="unbounded">
+        <xs:element ref="category"/>
+        <xs:element ref="sep"/>
+        <xs:element ref="label"/>
+        <xs:element ref="button"/>
+        <xs:element ref="block"/>
+      </xs:choice>
+      <xs:attribute name="name" use="required"/>
+      <xs:attribute name="expanded" type="xs:boolean"/>
+      <xs:attribute name="colour"/>
+      <xs:attribute name="categorystyle"/>
+      <xs:attribute name="custom"/>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="sep">
+    <xs:complexType>
+      <xs:attribute name="gap" type="xs:integer"/>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="label">
+    <xs:complexType>
+      <xs:attribute name="type" use="required" type="xs:NCName"/>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="button">
+    <xs:complexType>
+      <xs:attribute name="text" use="required"/>
+      <xs:attribute name="callbackKey" use="required" type="xs:NCName"/>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="variables">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element ref="variable" maxOccurs="unbounded"/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+  <xs:element name="variable">
+    <xs:complexType>
+      <xs:simpleContent>
+        <xs:extension base="xs:NCName">
+          <xs:attribute name="id"/>
+          <xs:attribute name="type" use="required"/>
```

Comment:
Note: I think this means old xml formats from before types are won't pass as valid..

Reply:
Currently we throw an error if we try to load a variable that doesn't have a type attribute:
`Error: Variable with id, -~F#HnYc~^iM+NrU/(#S is without a type xml.js:579:13`

Reply:
Oh, wait, I'm confused. This only applies to the list of variables at the top, which has always required a type. variableType is optional for fields.